### ISSUE TITLE
Re-Code

### ISSUE THREAD

**Issue by sekwah41** [sekwah41/Advanced-Portals#360]
Here are the tasks needed for the 
Ok so I'm going to make a list of features planned for the initial release. Please if anything is missing that you would consider core ping me or respond.

**This is what is considered as the core features, so some may be missing and replaced or added as they were later**

- [x] Command tags
- [x] Trigger block tags
- [x] Destination tags
- [x] Sort all command permissions and tag permissions surrounding testing.
- [x] Message tag
- [x] Add default warp message and config to say where they should show
- [x] Delay tag
- [x] Disable gateway beacons in portal regions
- [x] Bungee tag
- [x] Proxy/Server tag
- [x] Permission tag
- [x] Portal protection
- [x] Portal disable liquid physics
- [x] Portal disable mob spawning
- [x] Nether portal checks (avoid going to nether)
- [x] Fire portals don't keep you set on fire
- [x] Gateway blocks and end portals don't trigger default behaviour
- [x] Destination create
- [x] Destination remove
- [x] Portal create
- [x] Portal remove
- [x] Portal file conversion to new format
- [x] Portal selector
- [x] Placeholder blocks for portals, end and gateway
- [x] Help Menus
- [x] Stop portal blocks being smashed in active portals
- [x] Update permissions in the plugin.yml

These are the indented platforms to support at least eventually

Core

- Spigot
- Paper (possibly using some of the extra optimisations, plus properly testing paper specific bugs)
- Velocity
- Bungee

Extended
- Folia
- Forge
- Fabric

Will likely switch the non base release tasks to a separate ticket

**Comment by sekwah41** [comment]
The plan is for now to get the main system out, some parts may be rough and I will re-organise it as I go but the main API's or data used by plugins should not change drastically.

**Comment by sekwah41** [comment]
The re-code will be based off the initial re-code but with a slight re-structure.

**Comment by sekwah41** [comment]
Look into the best way to use/setup guice https://stackoverflow.com/questions/3404871/guice-inject-field-in-class-not-created-by-guice

For now the injection method seems a bit best for just getting it working!

**Comment by Terrorwolf01** [comment]
Hi,
Is there any Progress on this rewrite?
BR

**Comment by sekwah41** [comment]
Not in the last few weeks though I have the base tag system and most things ready with a few bugs for spigot. I just need to add bungee and velocity support. Then it's bug testing and finishing off a few more of the base tags.

**Comment by sekwah41** [comment]
Ok so I'm going to make a list of features planned for the initial release. Please if anything is missing that you would consider core ping me or respond.

- [ ] Command tags
- [ ] Trigger block tags
- [ ] Destination tags
- [ ] Bungee tags
- [ ] Bungee support
- [ ] Velocity support
- [ ] Portal protection
- [ ] Portal disable liquid physics
- [ ] Portal disable mob spawning
- [ ] Nether portal checks (avoid going to nether)
- [ ] Fire portals don't keep you set on fire
- [ ] Gateway blocks and end portals don't trigger default behaviour
- [ ] Destination create
- [ ] Destination remove
- [ ] Portal create
- [ ] Portal remove
- [ ] Portal file conversion to new format
- [ ] Portal selector
- [ ] Placeholder blocks for portals, end and gateway
- [ ] help menus

**Comment by sekwah41** [comment]
Ive added the extra checks to the main body of the ticket.

**Comment by TreemanKing** [comment]
Error when reloading an invalid configuration
```
Caused by: com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated array at line 27 column 5 path $.a.[2]
```

**Comment by TreemanKing** [comment]
I believe better error logging will prevent most Nether checks as currently, if there is an error somewhere in the configuration or creation, it will send you there.

**Comment by TreemanKing** [comment]
Cooldown does not work. No error message as well.

**Comment by sekwah41** [comment]
Once this is ready I need to make an announcement saying some features may be slightly different or missing, so please raise issues or discuss on the discord.

**Comment by sekwah41** [comment]
Need to fix an issue with creating tags using " to surround values

**Comment by sekwah41** [comment]
Will merge tomorrow, and after testing will look to release a full update

**Comment by sekwah41** [comment]
Actually I'll keep open as I'd like to also look at moving dependencies via the shadow plugin so they don't clash. Also I need to setup the release pipeline.

**Comment by sekwah41** [comment]
Closing now as ive remapped and shadowed the libraries
